### PR TITLE
Jenish/feature/logger

### PR DESCRIPTION
**Pull request by JenishPatel99**

### CODE REVIEW

File: Core/Filesystem/test/nandfs_tests.c [hunk 1]
```diff
@@ -79,16 +79,24 @@ int check_all_pages_in_block(int block, char c) {
     }
 }
 
-int write_all_pages_in_block(int block, char c) {
+int read_from_block_one(uint16_t page) {
```
Comment (Taylorr82) [inline Core/Filesystem/test/nandfs_tests.c:82]: Why did you delete the other test?

File: Core/Src/command_handler.c [hunk 2]
```diff
@@ -218,6 +230,7 @@ int onboot_sensors(uint8_t sensor) {
 
     if (vid != 0x56 || pid != 0x42) {
         DBG_PUT("Camera %x I2C Address: Unknown\r\nVIS not available\r\n\n", sensor);
```
Comment (Taylorr82) [inline Core/Src/command_handler.c:232]: Remove old DBG_PUT

File: Core/Src/command_handler.c [hunk 3]
```diff
@@ -306,6 +320,7 @@ int transfer_image_to_nand(uint8_t sensor, uint8_t *file_timestamp) {
     spi_init_burst(sensor);
     for (int j = 0; j < chunks_to_write; j++) {
         DBG_PUT("Writing chunk %d / %d\r\n", j + 1, chunks_to_write);
+        // iris_log("Writing chunk %d / %d", j + 1, chunks_to_write);
```
Comment (Taylorr82) [inline Core/Src/command_handler.c:323]: Commented code

File: Core/Src/logger.c [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+#include "nand_types.h"
+#include "nandfs.h"
+#include "nand_m79a_lld.h"
+#include "debug.h"
+#include "iris_time.h"
+#include "command_handler.h"
+
+extern SPI_HandleTypeDef hspi1;
+
+uint8_t logger_buffer[PAGE_DATA_SIZE];
```
Comment (Taylorr82) [inline Core/Src/logger.c:23]: It would be cleaner to combine all these in to a struct and pass the pointer to that struct to your functions

File: Core/Src/logger.c [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+        NAND_ReturnType ret = NAND_Block_Erase(&addr);
+        if (ret != Ret_Success) {
+            DBG_PUT("Error in deleting block b %d p %d r %d\r\n", addr.block, addr.page, ret);
+            return ret;
+        }
+    }
+    return 0;
+}
+
+int read_from_block(uint8_t block, uint16_t page) {
```
Comment (Taylorr82) [inline Core/Src/logger.c:145]: Is this a test function? It doesn't do anything with the data other than DBG_PUT it

File: Core/Src/main.c [hunk 6]
```diff
@@ -768,6 +770,12 @@ static void onboot_commands(void) {
     DBG_PUT("-----------------------------------\r\n");
 #endif
 #endif
+
+    HAL_Delay(1000);
```
Comment (Taylorr82) [inline Core/Src/main.c:774]: Why delay on boot?

File: Core/Src/obc_handler.c [hunk 7]
```diff
@@ -79,10 +81,13 @@ int obc_handle_command(uint8_t obc_cmd) {
         uint8_t buffer[sizeof(hk)];
         memcpy(buffer, &hk, sizeof(hk));
         obc_spi_transmit(buffer, sizeof(buffer));
+        iris_log("Done sending housekeeping data");
         return 0;
     }
     case IRIS_TAKE_PIC: {
         take_image();
+        DBG_PUT("Image capture complete");
```
Comment (Taylorr82) [inline Core/Src/obc_handler.c:89]: iris_log should send the logs to DBG_PUT if configured to do so, instead of having two calls with the same string

File: Core/Src/obc_handler.c [hunk 8]
```diff
@@ -115,9 +120,16 @@ int obc_handle_command(uint8_t obc_cmd) {
 #endif
         return 0;
     }
+    case IRIS_TRANSFER_LOG: {
+        clear_and_dump_buffer();
```
Comment (Taylorr82) [inline Core/Src/obc_handler.c:124]: This sequence will cause the rest of a page to be wasted if the logger doesn't write the whole page in this dump. Does it handle this situation? 

Also, the first thing transfer_log_to_obc() does is dump the buffer

File: Core/Filesystem/test/nandfs_tests.c (hunk 1 above)
Comment (JenishPatel99) [reply Core/Filesystem/test/nandfs_tests.c:82]: Hmm, do not know why I did that, nevertheless I have added it back

File: Core/Src/command_handler.c (hunk 2 above)
Comment (JenishPatel99) [reply Core/Src/command_handler.c:233]: Refer to #65

File: Core/Src/logger.c (hunk 5 above)
Comment (JenishPatel99) [reply Core/Src/logger.c]: I have moved this to the nand testing directory

File: Core/Src/command_handler.c (hunk 3 above)
Comment (JenishPatel99) [reply Core/Src/command_handler.c]: Removed

File: Core/Src/main.c (hunk 6 above)
Comment (JenishPatel99) [reply Core/Src/main.c]: Probably leftovers from debugging

**Review by Taylorr82 — APPROVED** [review]
